tools/plugins/com.liferay.ide.project.ui/src/com/liferay/ide/project/ui/quickfix/LiferayDependencyQuickFix.java

```diff
@@ -0,0 +1,269 @@
+                        createDepProposal( context, proposals, bundle );
+                    }
+                }
+            }
+        }
+        catch( Exception e )
+        {
+            ProjectCore.logError( "Add module dependence got error", e );
+        }
+    }
+
+    private void createDepProposal(
+        IInvocationContext context, Collection<IJavaCompletionProposal> proposals, ServiceContainer bundle )
+    {
+        final String bundleGroup = bundle.getBundleGroup();
+        final String bundleName = bundle.getBundleName();
+        final String bundleVersion = bundle.getBundleVersion();
+        proposals.add(
+            new CUCorrectionProposal( "Add Liferay Dependence " + bundleName, context.getCompilationUnit(), null, -0)
+            {
+
+                @Override
+                public void apply( IDocument document )
+                {
+                    try
+                    {
+                        IJavaProject javaProject = context.getCompilationUnit().getJavaProject();
+                        IProject project = javaProject.getProject();
+
+                        ILiferayProject liferayProject = LiferayCore.create( project );
+                        final IProjectBuilder builder = liferayProject.adapt( IProjectBuilder.class );
+
+                        if( builder != null )
+                        {
+                            Version retriveVersion = new Version(bundleVersion);
+
+                            String[] dependency = new String[] { bundleGroup, bundleName, retriveVersion.getMajor() + "." + retriveVersion.getMinor() + ".0" };
+                            List<String[]> dependencyList = new ArrayList<String[]>();
+                            dependencyList.add( dependency );
+                            builder.updateProjectDependency( project, dependencyList );
+                        }
+                    }
+                    catch( Exception e )
+                    {
+                        ProjectUI.logError( "Adding Liferay Module dependence got error", e );
+                    }
+                }
+
+                @Override
+                public Object getAdditionalProposalInfo( IProgressMonitor monitor )
+                {
+                    return "Add dependenece";
+                }
+
+                @Override
+                public Image getImage()
+                {
+                    Display display = UIUtil.getActiveShell().getDisplay();
+                    String file = null;
+
+                    try
+                    {
+                        file = FileLocator.toFileURL(
+                            ProjectUI.getDefault().getBundle().getEntry( "icons/e16/liferay_logo_16.png" ) ).getFile();
+                    }
+                    catch( IOException e )
+                    {
+                        ProjectUI.logError( e );
+                    }
+
+                    return new Image( display, file );
```

Comment:
Use `ImageDescriptorFromURL` method to get this image instead of creating this image each time (which will run out of resources eventually)  Use the Imageregistry pattern